check-style.sh

```diff
@@ -0,0 +1,136 @@
+        # these.
+        affected_files=$(git diff --cached --name-only --diff-filter=ACM HEAD)
+        ;;
+    * )
+        echo "Unknown argument: $1"
+        print_help
+        exit 2
+esac
+
+
+# Unset variable would be a sign of programmer error. We are not using '-e' in
+# this script as we'd like to handle these cases ourselves where relevant, i.e.,
+# allow more than one code check failure per run.
+set -u
+
+# Work from the git root. This is important to help some tools pick up the
+# correct configuration.
+git_root=$(git rev-parse --show-toplevel)
+cd "${git_root}"
+
+# Define a cummulative status code for the script. The value is Updated through
+# 'run_check' when running checks. A status code creater than 0 will indicate
+# that one or more checks failed.
+status_code=0
+
+# Run a code check command and update the cummulative error code.
+run_check() {
+    $@
+    status_code=$(($status_code + $?))
+}
+
+# Helper function to filter affected files by extension.
+get_files_by_extension() {
+    filter=""
+    for ext in $@; do
+        # We want a `.` to be a full stop, not a regexp wildcard.
+        filter="$(echo $ext | sed -e 's|^\.|\\.|')$|${filter}"
+    done
+    filter=$(echo $filter | sed -e 's/|$//')
+
+    echo $affected_files | tr ' ' '\n' | egrep $filter
+}
+
+# Check files that we can clang-format.
+clang_format_files=$(get_files_by_extension .cc .cpp .h .hpp .proto)
+if [ ! -z "${clang_format_files}" ]; then
+    # Run clang-format. In fact, run our wrapper script around-clang format that
+    # also contains logic for checking the line length.
+    #
+    # ISSUE https://github.com/reboot-dev/respect/issues/1371: This script is
+    # very slow as we process each file sequentially and does the line length
+    # checking in bash.
+    #
+    # Depending on whether we are in pre-commit mode of full mode, we want to
+    # invoke these scripts differently:
+    # TODO: Consider tidying up this and perhaps solve #1371 in the process.
+    dev_tools_path=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+    case $mode in
+        "full" )
+            # For full checking, we can invoke the wrapper script directly.
+            run_check "${dev_tools_path}/check-code-style/check_style_of_all_files.sh"
+        ;;
+        "pre-commit" )
+            # For the pre-commit check we must source another bash script and
+            # then we have a function available to us...
+            source "${dev_tools_path}/check-code-style/check_style.sh"
+            run_check check_style_of_files_in_commit
+        ;;
+        * )
+            # For safety catch these.
+            echo "Unsupported mode: $mode. This should not happen."
+            exit 1
+        ;;
+    esac
+fi
+
+# Check bazel files
+bazel_files=$(get_files_by_extension .bzl .bazel BUILD WORKSPACE)
+if [ ! -z "${bazel_files}" ]; then
+    run_check buildifier --lint=warn --warnings=all --mode=check
```

Comment:
Also [scope `buildifier`'s check down](https://github.com/bazelbuild/buildtools/blob/master/buildifier/README.md#usage) to just the `bazel_files`?